### PR TITLE
feat(ci): using a local private registry to transfer images from build server to test server(s)

### PR DESCRIPTION
**Pull request by rdefosse**
So far I did here the part pulling onto the test server.

Only the following NFs are impacted:

* HSS
* SPGWC
* SPGWU-TINY

I did not do `MAGMA-MME` because I don't want to expose it to MAGMA community yet. But I could push the daily `master` in the `MAGMA-MME-master-check` pipeline.

Validated on this run --> `https://jenkins-oai.eurecom.fr/view/CN4G/job/MAGMA-MME-U18-Deploy-Check-Dev/4/`

Also validated the no-impact on the RHEL8 one --> `https://jenkins-oai.eurecom.fr/view/CN4G/job/MAGMA-MME-RHEL8-Deploy-Check-Dev/1/`

Next I need to update the building pipelines for the above-mentioned  NFs to push to the registry.

### CODE REVIEW

**Comment by NeptuneMagicSauce** [comment]
What about the errors in the 2 Jenkins validating runs?

`MAGMA-MME-U18-Deploy-Check-Dev` and `MAGMA-MME-RHEL8-Deploy-Check-Dev` say 

`Connection error: ('Unable to connect to any servers', {'192.168.61.66': error(111, "Tried connecting to [('192.168.61.66', 9042)]. Last error: Connection refused")})` 

at step `python3 ./ci-scripts/OAI-RAN-Sanity-Check-Deploy.py --action=DeployCassandra`

`MAGMA-MME-U18-Deploy-Check-Dev` says 

`Error: No such image: oai-hss:develop-e1a0653c` 

and 2 more similar messages at step `docker rmi ...`

**Comment by rdefosse** [comment]
> `Connection error: ('Unable to connect to any servers', {'192.168.61.66': error(111, "Tried connecting to [('192.168.61.66', 9042)]. Last error: Connection refused")})`
> 
> at step `python3 ./ci-scripts/OAI-RAN-Sanity-Check-Deploy.py --action=DeployCassandra`
> 

As any database, it takes time for the tables and services to be ready. 

Lionel and I did not find any other way as try and see if we do not get any errors. 

So this is normal behavior

**Comment by rdefosse** [comment]
> `MAGMA-MME-U18-Deploy-Check-Dev` says
> 
> `Error: No such image: oai-hss:develop-e1a0653c`
> 
> and 2 more similar messages at step `docker rmi ...`

Because I do it twice:

* Once in the normal undeployment stage
* Second time in the post `always` stage

In CI better be redundant than sorry :)

**Review by NeptuneMagicSauce — APPROVED** [review]

File: ci-scripts/Jenkinsfile-Magma-Sanity-Check [hunk 1]
```diff
@@ -141,6 +141,54 @@ pipeline {
...
             }
           }
+
+          // Prepare workspace
+          sh './scripts/syncComponents.sh --hss-branch ' + hssBranch + ' --spgwc-branch ' + spgwcBranch + ' --spgwu-tiny-branch ' + spgwuBranch
+
+          // On Ubuntu servers, we shall pull from private local registry
+          if ((cn_ci_host == 'selfix') || (cn_ci_host == 'orion')) {
+            // Login
+            sh containerCli + ' login -u oaicicd -p oaicicd selfix.sboai.cs.eurecom.fr > /dev/null 2>&1'
```
Comment (arora-sagar) [inline ci-scripts/Jenkinsfile-Magma-Sanity-Check:151]: @rdefosse There is no possibility to mask the password or save it as Jenkins secret?

Comment (rdefosse) [reply ci-scripts/Jenkinsfile-Magma-Sanity-Check:151]: @arora-sagar 

we could but on the RAN, we kept it as open like this.

You need to be inside the Eurecom infra to do anything anyway.

**Pull request merged**